pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -739,28 +743,31 @@ func getDestinationRule(push *model.PushContext, proxy *model.Proxy, hostname ho
 	return nil, nil
 }
 
-func hasOutlierDetection(push *model.PushContext, proxy *model.Proxy, clusterName string) bool {
+func getOutlierDetectionAndLoadBalancerSettings(push *model.PushContext, proxy *model.Proxy, clusterName string) (bool, *networkingapi.LoadBalancerSettings) {
 	_, subsetName, hostname, portNumber := model.ParseSubsetKey(clusterName)
 
+	var lbSettings *networkingapi.LoadBalancerSettings
 	destinationRule, port := getDestinationRule(push, proxy, hostname, portNumber)
 	if destinationRule == nil || port == nil {
-		return false
+		return false, nil
 	}
 
-	_, outlierDetection, _, _ := networking.SelectTrafficPolicyComponents(destinationRule.TrafficPolicy, port)
+	_, outlierDetection, loadBalancerSettings, _ := networking.SelectTrafficPolicyComponents(destinationRule.TrafficPolicy, port)
+	lbSettings = loadBalancerSettings
 	if outlierDetection != nil {
-		return true
+		return true, loadBalancerSettings
```

Comment:
This is not right. The loadBalancerSettings orders should be subset.TrafficPolicy --> destinationrule.TrafficPolicy --> global mesh config. But here, you reversed the first two orders.

Reply:
Good catch, updated.